jsonrpc/eth_endpoint.go

```diff
@@ -220,6 +220,11 @@ func (e *Eth) Call(arg *txnArgs, number BlockNumber) (interface{}, error) {
 		return nil, err
 	}
 
+	// If the caller didn't supply the gas limit in the message, then we set it to maximum possible => block gas limit
+	if transaction.Gas == 0 {
+		transaction.Gas = header.GasLimit
+	}
```

Comment:
Yep, the reason why we set it to the gasLimit is because eth_call would fail the intrinsic gas checks before executing (since it is basically simulating a transctions).